Project/Source/Components/ComponentParticleSystem.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+#include "Component.h"
+
+#include "Utils/Pool.h"
+#include "Utils/UID.h"
+#include "Math/float4.h"
+#include "Math/float2.h"
+#include "Math/float4x4.h"
+#include "Math/Quat.h"
+
+#include <vector>
+class ComponentTransform;
+class ParticleModule;
+
+enum class EmitterType {
+	CONE,
+	SPHERE,
+	HEMISPHERE,
+	DONUT,
+	CIRCLE,
+	RECTANGLE
+};
+
+class ComponentParticleSystem : public Component {
+public:
+	struct Particle {
+		float4x4 model;
+
+		float3 initialPosition;
+		float3 position;
+		float3 direction;
+		float3 scale;
+
+		Quat rotation;
+
+		float velocity;
+		float life = 0.0F;
+		float currentFrame;
+		float colorFrame;
```

Comment:
Variables should be initialized. If you want to avoid initializations you can add a constructor that initializes the viariables.